mantidimaging/gui/windows/main/model.py

```diff
@@ -91,18 +91,6 @@ def do_nexus_saving(self, dataset_id: uuid.UUID, path: str, sample_name: str):
         else:
             raise RuntimeError(f"Failed to get StrictDataset with ID {dataset_id}")
 
-    def set_image_data_by_uuid(self, images_id: uuid.UUID, new_stack: ImageStack) -> None:
-        """
-        Updates the data of an existing dataset/images object.
-        :param images_id: The id of the image to update.
-        :param new_stack: The ImageStack containing the new data.
-        """
-        for dataset in self.datasets.values():
-            if images_id in dataset:
-                dataset.replace(images_id, new_stack)
```

Comment:
I don't think this `replace` method from `BaseDataset` is used anywhere now. Would it be worth removing, or do we think it's worth keeping in case it's needed in the future?